src/mgmt/tests/mocks.c

```diff
@@ -168,6 +168,106 @@ mock_ds_mgmt_pool_delete_acl_teardown(void)
 	daos_acl_free(ds_mgmt_pool_delete_acl_result);
 }
 
+int				ds_mgmt_list_pools_return;
+char				ds_mgmt_list_pools_group[DAOS_SYS_NAME_MAX];
+void				*ds_mgmt_list_pools_npools_ptr;
+uint64_t			ds_mgmt_list_pools_npools;
+void				*ds_mgmt_list_pools_poolsp_ptr;
+struct mgmt_list_pools_one	*ds_mgmt_list_pools_poolsp_out;
+void				*ds_mgmt_list_pools_len_ptr;
+size_t				ds_mgmt_list_pools_len_out;
+int
+ds_mgmt_list_pools(const char *group, uint64_t *npools,
+		   struct mgmt_list_pools_one **poolsp, size_t *pools_len)
+{
+	size_t i;
+
+	strncpy(ds_mgmt_list_pools_group, group, DAOS_SYS_NAME_MAX);
+
+	ds_mgmt_list_pools_npools_ptr = (void *)npools;
+	if (npools != NULL)
+		ds_mgmt_list_pools_npools = *npools;
+
+	ds_mgmt_list_pools_poolsp_ptr = (void *)poolsp;
+	if (poolsp != NULL && ds_mgmt_list_pools_poolsp_out != NULL) {
+		size_t	pool_array_size;
+
+		pool_array_size = ds_mgmt_list_pools_len_out *
+				  sizeof(struct mgmt_list_pools_one);
+
+		D_ALLOC(*poolsp, pool_array_size);
```

Comment:
can this be reduced to a D_ALLOC_ARRAY() without need to compute pool_array_size?

Reply:
Good catch, this was lingering from one of my earlier implementation attempts.